### DISCUSSION TITLE
Allow the DesignerSerializationVisibilityAttribute to be used on a class?

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/csharplang#7057] [Polls]
The option to also add the  `DesignerSerializationVisibilityAttribute` on a class should be implemented to avoid having to type it multiple times where that class is used as a property. Simply by adding in the `AttributeUsage`  to also include  `AttributeTargets.Class` enum.

**Comment by 333fred** [comment]
It looks like this is an api request: please open an issue on the dotnet/runtime repo for this.
